projects/fal/src/fal/app.py

```diff
@@ -300,7 +359,16 @@ async def set_global_object_preference(request, call_next):
                     "Failed set a global lifecycle preference %s",
                     self.__class__.__name__,
                 )
-            return await call_next(request)
+
+            request_id = request.headers.get(REQUEST_ID_KEY)
```

Comment:
Let's make this a new middleware rather than using global object prefernce, with the correct try catch, etc

Reply:
oh wow, haha